File: data_parse.pl
#!/usr/local/bin/perl

my $player = "";
my $year = 0;
my $month = 0;
my $day = 0;
my $kaisen = 0;
my $flag_point = 0;

while (<>){
	$_ =~ s/\r//g;
	$_ =~ s/\n//g;

	if ($flag_point == 1){
		$_ =~ s/ //g;
		$_ =~ s/\t//g;
		$_ =~ s/▲/-/g;	
		$_ =~ s/pt//;
		$_ =~ s/\(.*\)//; #チョンボ対策
		$point = $_;

		if ($year >= 2021){
			printf("%04d%02d%02d%02d,%s,%s\n", $year, $month, $day, $kaisen, $player, $point);
		}
		$flag_point = 0;
		next;
	}

	if (/<div class=\"p-gamesResult__date\">(\d+)\/(\d+)</){
		$month = $1;
		$day = $2;
		if ($month >= 10){
			$year = 2021;
		}
		else {
			$year = 2022;
		}
		next;
	}

	if (/<div class=\"p-gamesResult__number\">[^0-9]+([0-9])[^0-9]+/){
		$kaisen = $1;
	}

	if (/<div class=\"p-gamesResult__name\">(.+)<\/div>/){
		$player = $1;
		next;
	}

	if (/<div class=\"p-gamesResult__point\">/){
		$flag_point = 1;
		next;
	}
}

File: data_parse.py
#!/usr/bin/python3 -u

import sys,os,re
import fileinput
player=''
year=0
month=0
day=0
kaisen=0
flag_point=0

for buf in sys.stdin:
    buf = re.sub('\r','',buf)
    buf = re.sub('\n','',buf)

    if flag_point==1:
        buf = re.sub(' ','',buf)
        buf = re.sub('\t','',buf)
        buf = re.sub('▲','-',buf)
        buf = buf.replace('pt','',1)
        buf = re.sub('\(.*\)','',buf)
        point=buf

        if year>=2021:
            print("%04d%02d%02d%02d,%s,%s" % (year,month,day,kaisen,player,point))
        flag_point=0
        continue 

    buf_search=re.search('<div class=\"p-gamesResult__date\">(\d+)\/(\d+)<',buf)
    if buf_search:
        month=int(buf_search.group(1))
        day=int(buf_search.group(2))
        if month>=10:
            year=2021
        else:
            year=2022
        continue 

    buf_search=re.search('<div class=\"p-gamesResult__number\">[^0-9]+([0-9])[^0-9]+',buf)
    if buf_search:
        kaisen=int(buf_search.group(1))

    buf_search=re.search('<div class=\"p-gamesResult__name\">(.+)<\/div>',buf)
    if buf_search:
        player=buf_search.group(1)
        continue 

    if re.search('<div class=\"p-gamesResult__point\">',buf):
        flag_point=1
        continue 
